app/models/spree/order_decorator.rb

```diff
@@ -398,9 +398,11 @@ def ensure_customer
 
   def update_adjustment!(adjustment)
     return if adjustment.finalized?
+
     state = adjustment.state
     adjustment.state = 'open'
     adjustment.update!
+    update!
```

Comment:
since manually opening the adjustment makes the test pass, this means that in v2 the `OrderUpdater` (the class that updates all the totals attributes) is not executed before the adjustment's state is restored in line 406. Therefore, triggering the update ourselves fixes it.